src/Test.ConcurrencyTests/BasicCoyoteTests.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Threading.Tasks;
+
+using Microsoft.Coyote;
+using Microsoft.Coyote.Specifications;
+using Microsoft.Coyote.SystematicTesting;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Test.CoyoteTests
+{
+    [TestClass]
+    public class BasicCoyoteTests
+    {
+        /// <summary>
+        /// This is a very simple concurrency unit test, where the bug is hard to manifest
+        /// using traditional test infrastructure. This test serves as a template for using
+        /// Coyote testing (which is able to rapidly reveal the assertion failure).
+        /// </summary>
+        /// <returns></returns>
+        [TestMethod]
+        public async Task TestTaskAsync()
+        {
+            int value = 0;
+            Task task = Task.Run(() =>
+            {
+                value = 1;
+            });
+
+            Specification.Assert(value == 0, "value is 1");
+            await task;
+        }
+
+        [TestMethod, TestCategory("NightlyTest")]
```

Comment:
`NightlyTest` is the new category of tests which will be selectively run on a nightly basis. As @shaopeng-gh suggested, the category name is deliberately kept generic, in case we want to run non-Coyote tests on a nightly basis as well.

Reply:
just throwing some names for consideration 
"CI"
"Integration"

Reply:
@shaopeng-gh I am not in favor of either CI or Integration, but look forward to more thoughts on this.

Reply:
see if we want to define something so no string typo difference:
TestCategories.Unit
TestCategories.XXX

Reply:
@shaopeng-gh Sorry I didn't fully understand this. Could you kindly give me an example?

Reply:
oh nothing important, I just mean use plain string or not:
`[TestMethod, TestCategory("Nightly")]`
`[TestMethod, TestCategory(TestCategories.Nightly)]`